src/java/voldemort/store/readonly/ReadOnlyStorageConfiguration.java

```diff
@@ -48,7 +49,18 @@ public ReadOnlyStorageConfiguration(VoldemortConfig config) {
         this.nodeId = config.getNodeId();
         this.deleteBackupMs = config.getReadOnlyDeleteBackupMs();
         this.maxValueBufferAllocationSize = config.getReadOnlyMaxValueBufferAllocationSize();
-        this.voldConfig = config;
+
+        FailedFetchLock failedFetchLockInstance;
+        if (config.getHighAvailabilityStateAutoCleanUp()) {
+            try {
+                failedFetchLockInstance = FailedFetchLock.getLock(config, null);
```

Comment:
HdfsFailedFetchLock initializes the HadoopFileSystem on the constructor. The HadoopFileSystem opened on the constructor is prone to connection resets and failures. Also this will force all the store to have an open handle to the HdfsFileSystem which could cause more issues.